### PR TITLE
21 delete the top up functionality from the contracts

### PR DESCRIPTION
**Pull request by Vitomir2**
- delete the top-up functionality, rename errors and variables, and adapt tests;
- add some additional tests for the getDelegatorPositionReward and swap balance change index;

### CODE REVIEW

File: contracts/ValidatorSet/modules/Delegation/IDelegation.sol [hunk 1]
```diff
@@ -20,22 +20,13 @@ interface IDelegation {
 
     /**
      * @notice Delegates sent amount to validator. Set vesting position data.
-     * Delete old top-ups data if exists. Can be called by vesting positions' managers only.
+     * Delete old pool params data, if exists. Can be called by vesting positions' managers only.
```
Comment (R-Santev) [inline contracts/ValidatorSet/modules/Delegation/IDelegation.sol:23]:
```suggestion
     * Delete old pool params data, if exists.
```

The deleted part is explained on the next line

File: test/ValidatorSet/Staking.test.ts [hunk 2]
```diff
@@ -220,8 +220,8 @@ export function RunStakingTests(): void {
       });
     });
 
-    describe("Top-up staking position with stake()", function () {
-      it("should top-up staking position", async function () {
+    describe("Increase staking position using stake()", function () {
```
Comment (R-Santev) [inline test/ValidatorSet/Staking.test.ts:223]: staking top-up must be removed as well

File: test/ValidatorSet/ValidatorSet.test.ts [hunk 3]
```diff
@@ -830,35 +830,6 @@ describe("ValidatorSet", function () {
         );
       });
 
-      it("should emit StakeChanged event on top-up vested position", async function () {
```
Comment (R-Santev) [inline test/ValidatorSet/ValidatorSet.test.ts:833]: Such test must be made for the swap

**Review by R-Santev — CHANGES_REQUESTED** [review]

**Comment by Vitomir2** [comment]
Delete the stake top-up and polished

**Review by R-Santev — APPROVED** [review]

**Pull request merged**